NoClassDefFoundError with Spring 5

With Spring framework 5.0.0, SAMLBootstrap class throws NoClassDefFoundError.

```
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.security.saml.SAMLBootstrap]: No default constructor found; nested exception is java.lang.NoClassDefFoundError: org/springframework/beans/factory/access/BootstrapException
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:84)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateBean(AbstractAutowireCapableBeanFactory.java:1223)
	... 59 common frames omitted
Caused by: java.lang.NoClassDefFoundError: org/springframework/beans/factory/access/BootstrapException
	at java.lang.Class.getDeclaredConstructors0(Native Method)
	at java.lang.Class.privateGetDeclaredConstructors(Class.java:2671)
	at java.lang.Class.getConstructor0(Class.java:3075)
	at java.lang.Class.getDeclaredConstructor(Class.java:2178)
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:79)
	... 60 common frames omitted
Caused by: java.lang.ClassNotFoundException: org.springframework.beans.factory.access.BootstrapException
	at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1285)
	at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1119)
	... 65 common frames omitted
```


The BootstrapException class seems to be removed in Spring 5.

--- a/core/src/main/java/org/springframework/security/saml/SAMLBootstrap.java
+++ b/core/src/main/java/org/springframework/security/saml/SAMLBootstrap.java
@@ -20,7 +20,7 @@ import org.opensaml.xml.ConfigurationException;
 import org.opensaml.xml.security.keyinfo.NamedKeyInfoGeneratorManager;
 import org.opensaml.xml.security.x509.X509KeyInfoGeneratorFactory;
 import org.springframework.beans.BeansException;
-import org.springframework.beans.factory.access.BootstrapException;
+import org.springframework.beans.FatalBeanException;
 import org.springframework.beans.factory.config.BeanFactoryPostProcessor;
 import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
 
@@ -42,7 +42,7 @@ public class SAMLBootstrap implements BeanFactoryPostProcessor {
             PaosBootstrap.bootstrap();
             setMetadataKeyInfoGenerator();
         } catch (ConfigurationException e) {
-            throw new BootstrapException("Error invoking OpenSAML bootstrap", e);
+            throw new FatalBeanException("Error invoking OpenSAML bootstrap", e);
         }
     }
 

